### ISSUE TITLE
Warnings generated

### ISSUE THREAD

**Issue by jkammerland** [Clemapfel/jluna#40]
with GCC 12 and clang 15 the code  
```
initialize();
Base["println"]("hello julia");

/// declare function
Main.safe_eval("f(x) = x^x");

// bind function to C++-side variable
auto f = Main.safe_eval("return f");

// call function
Int64 result = f(3);

// print result
std::cout << result << std::endl;

return 0;
```
Produce these warnings before program output:

```
warning: copy relocation against non-copyable protected symbol `jl_int32_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_float32_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_float64_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_int64_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_base_module' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_bool_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_uint64_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_uint16_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_int16_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_float16_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_uint32_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_int8_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_uint8_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
warning: copy relocation against non-copyable protected symbol `jl_char_type' in `/home/jonkam/Programmering/julia/julia-1.8.5-linux-x86_64/julia-1.8.5/lib/libjulia.so.1'
[JULIA][LOG] initialization successful (1 thread(s)).
hello julia
27

Process finished with exit code 0
```
Do I need to add some compiler flags ?

**Comment by Clemapfel** [comment]
I can reproduce this on master, this is related to a C-API change in Julia 1.8+, warning does not appear with the 1.7.x releases.

I'll investigate but I don't think this hints at an actual bug, it's just bad practice to force the copy of isbitstype memory

**Comment by Clemapfel** [comment]
MWE:
```cpp
int main()
{
    initialize();
    auto proxy = Proxy(jl_nothing);
    return 0;
}
```
```
warning: copy relocation against non-copyable protected symbol `jl_nothing' in `/lib64/libjulia.so.1'
[JULIA][LOG] initialization successful (1 thread(s)).

Process finished with exit code 0
```

**Comment by Clemapfel** [comment]
c.f. https://github.com/JuliaLang/julia/issues/48699

**Comment by Clemapfel** [comment]
Adding

```cmake
target_compile_options(<your_executable> PRIVATE "-fpic")
```

Fixed it for me. Can you confirm this? `<your_executable>` is your library or executable, not jluna.

**Comment by Clemapfel** [comment]
Fixed by https://github.com/Clemapfel/jluna/pull/39/commits/4c21434b965567223687585e0c9839d521a8a92a

I merged it into the 1.0.0 PR because I want to test this on another machine, I'm not positive that `-fpic` is supported by all compiler versions

**Comment by Clemapfel** [comment]
Thank you for confirming, I added a section to the troubleshooting section, c.f. https://github.com/Clemapfel/jluna/commit/e55427170f5e3a695669926895753600bdaaeef2
